shared/gcs/gcs.go

```diff
@@ -19,96 +19,154 @@ limitations under the License.
 package gcs
 
 import (
-	"bufio"
 	"context"
-	"fmt"
 	"io/ioutil"
 	"log"
-	"strconv"
+	"path"
+	"os"
+	"io"
 	"strings"
 
 	"cloud.google.com/go/storage"
 	"google.golang.org/api/option"
-)
-
-const (
-	bucketName = "knative-prow"
-	latest     = "/latest-build.txt"
+	"google.golang.org/api/iterator"
 )
 
 var client *storage.Client
 
-func createStorageClient(ctx context.Context, sa string) error {
+// Authenticate explicitly sets up authentication for the rest of run
+func Authenticate(ctx context.Context, serviceString string) error {
```

Comment:
"serviceString" sounds unknown to me. I strongly suggest "serviceAccount", which originally "sa" means.

Reply:
fixed